### DISCUSSION TITLE
New FS-UAE fork

### DISCUSSION THREAD

**Discussion by grahambates** [prb28/vscode-amiga-assembly#238] [General]
I've been working on a [new FS-UAE fork](https://github.com/grahambates/fs-uae/tree/barto_prb) based on the current upstream version and your WinUAE fork (originally based on Bartman's). I think I pretty much have it working on Mac when running it using the WinUAE launch config. I'll continue to do some testing and look at getting cross-platform builds working.

This would solve the problem of having separate remote debugger implementations and being stuck on an old version of FS-UAE.  It would also open up the possibility of implementing profiling in this extension, which is something I'm keen to explore.

**Comment by prb28** [comment]
That would be a really great step forward.

**Comment by prb28** [comment]
@grahambates the unified configuration and new fs-uae look great.
I've noted two things with a quick test: 
- the debugger was not able to connect to fs-uae, the emulator started but the breakpoints are not hit
- when you debug 2 times with winuae, during the second sessions the sources are not resolved during debug
I'll try to debug the extension later this week to investigate these issues.

**Reply by grahambates** [reply]
I'll have a look at these issues this weekend. I suspect that they're both related to incorrect normalization of Windows filenames i.e. the path of the breakpoint source doesn't match what's in the source map.

**Comment by prb28** [comment]
@grahambates The disassemble file feature is broken: 
(I'll try to fix it later this week-end)
 1) Global Extension Tests
       Disassemble command
         Should disassemble a file:

      AssertionError: expected ' ori.b\t#$2c,d0; 0  00 00 00 2c  ori.…' to equal ' movea.l\t$4.w,a6; 0  2c 78 00 04  mo…'
      + expected - actual

      - ori.b	#$2c,d0; 0  00 00 00 2c  ori.b	#$2c, d0
      - ori.b	#$78,d0; 4  00 00 00 78  ori.b	#$78, d0
      - ori.b	#$0,d0; 8  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$4,d0; c  00 00 00 04  ori.b	#$4, d0
Error: 1 tests failed.
	at Object.run (d:\a\vscode-amiga-assembly\vscode-amiga-assembly\out\test\suite\index.js:32:15)
	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
      - ori.b	#$70,d0;10  00 00 00 70  ori.b	#$70, d0
      - ori.b	#$0,d0;14  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$20,d0;18  00 00 00 20  ori.b	#$20, d0
      - ori.b	#$3c,d0;1c  00 00 00 3c  ori.b	#$3c, d0
      - ori.b	#$0,d0;20  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$0,d0;24  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$3,d0;28  00 00 00 03  ori.b	#$3, d0
      - ori.b	#$e8,d0;2c  00 00 00 e8  ori.b	#$e8, d0
      - ori.b	#$22,d0;30  00 00 00 22  ori.b	#$22, d0
      - ori.b	#$3c,d0;34  00 00 00 3c  ori.b	#$3c, d0
      - ori.b	#$0,d0;38  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$1,d0;3c  00 00 00 01  ori.b	#$1, d0
      - ori.b	#$0,d0;40  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$2,d0;44  00 00 00 02  ori.b	#$2, d0
      - ori.b	#$2c,d0;48  00 00 00 2c  ori.b	#$2c, d0
      - ori.b	#$78,d0;4c  00 00 00 78  ori.b	#$78, d0
      - ori.b	#$0,d0;50  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$4,d0;54  00 00 00 04  ori.b	#$4, d0
      - ori.b	#$4e,d0;58  00 00 00 4e  ori.b	#$4e, d0
      - ori.b	#$ae,d0;5c  00 00 00 ae  ori.b	#$ae, d0
      - ori.b	#$ff,d0;60  00 00 00 ff  ori.b	#$ff, d0
      - ori.b	#$3a,d0;64  00 00 00 3a  ori.b	#$3a, d0
      - ori.b	#$23,d0;68  00 00 00 23  ori.b	#$23, d0
      - ori.b	#$c0,d0;6c  00 00 00 c0  ori.b	#$c0, d0
      - ori.b	#$0,d0;70  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$0,d0;74  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$1,d0;78  00 00 00 01  ori.b	#$1, d0
      - ori.b	#$9c,d0;7c  00 00 00 9c  ori.b	#$9c, d0
      - ori.b	#$43,d0;80  00 00 00 43  ori.b	#$43, d0
      - ori.b	#$fa,d0;84  00 00 00 fa  ori.b	#$fa, d0
      - ori.b	#$1,d0;88  00 00 00 01  ori.b	#$1, d0
      - ori.b	#$5c,d0;8c  00 00 00 5c  ori.b	#$5c, d0
      - ori.b	#$4e,d0;90  00 00 00 4e  ori.b	#$4e, d0
      - ori.b	#$ae,d0;94  00 00 00 ae  ori.b	#$ae, d0
      - ori.b	#$fe,d0;98  00 00 00 fe  ori.b	#$fe, d0
      - ori.b	#$68,d0;9c  00 00 00 68  ori.b	#$68, d0
      - ori.b	#$22,d0;a0  00 00 00 22  ori.b	#$22, d0
      - ori.b	#$40,d0;a4  00 00 00 40  ori.b	#$40, d0
      - ori.b	#$28,d0;a8  00 00 00 28  ori.b	#$28, d0
      - ori.b	#$29,d0;ac  00 00 00 29  ori.b	#$29, d0
      - ori.b	#$0,d0;b0  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$26,d0;b4  00 00 00 26  ori.b	#$26, d0
      - ori.b	#$4e,d0;b8  00 00 00 4e  ori.b	#$4e, d0
      - ori.b	#$ae,d0;bc  00 00 00 ae  ori.b	#$ae, d0
      - ori.b	#$fe,d0;c0  00 00 00 fe  ori.b	#$fe, d0
      - ori.b	#$62,d0;c4  00 00 00 62  ori.b	#$62, d0
      - ori.b	#$1e,d0;c8  00 00 00 1e  ori.b	#$1e, d0
      - ori.b	#$3c,d0;cc  00 00 00 3c  ori.b	#$3c, d0
      - ori.b	#$0,d0;d0  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$80,d0;d4  00 00 00 80  ori.b	#$80, d0
      - ori.b	#$3c,d0;d8  00 00 00 3c  ori.b	#$3c, d0
      - ori.b	#$3c,d0;dc  00 00 00 3c  ori.b	#$3c, d0
      - ori.b	#$ff,d0;e0  00 00 00 ff  ori.b	#$ff, d0
      - ori.b	#$ff,d0;e4  00 00 00 ff  ori.b	#$ff, d0
      - ori.b	#$3a,d0;e8  00 00 00 3a  ori.b	#$3a, d0
      - ori.b	#$39,d0;ec  00 00 00 39  ori.b	#$39, d0
      - ori.b	#$0,d0;f0  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$df,d0;f4  00 00 00 df  ori.b	#$df, d0
      - ori.b	#$f0,d0;f8  00 00 00 f0  ori.b	#$f0, d0
      - ori.b	#$1c,d0;fc  00 00 00 1c  ori.b	#$1c, d0
      - ori.b	#$33,d0;100  00 00 00 33  ori.b	#$33, d0
      - ori.b	#$fc,d0;104  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$7f,d0;108  00 00 00 7f  ori.b	#$7f, d0
      - ori.b	#$ff,d0;10c  00 00 00 ff  ori.b	#$ff, d0
      - ori.b	#$0,d0;110  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$df,d0;114  00 00 00 df  ori.b	#$df, d0
      - ori.b	#$f0,d0;118  00 00 00 f0  ori.b	#$f0, d0
      - ori.b	#$9a,d0;11c  00 00 00 9a  ori.b	#$9a, d0
      - ori.b	#$20,d0;120  00 00 00 20  ori.b	#$20, d0
      - ori.b	#$7a,d0;124  00 00 00 7a  ori.b	#$7a, d0
      - ori.b	#$1,d0;128  00 00 00 01  ori.b	#$1, d0
      - ori.b	#$52,d0;12c  00 00 00 52  ori.b	#$52, d0
      - ori.b	#$30,d0;130  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$fc,d0;134  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$1,d0;138  00 00 00 01  ori.b	#$1, d0
      - ori.b	#$fc,d0;13c  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$30,d0;140  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$fc,d0;144  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$0,d0;148  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$0,d0;14c  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$30,d0;150  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$fc,d0;154  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$1,d0;158  00 00 00 01  ori.b	#$1, d0
      - ori.b	#$0,d0;15c  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$30,d0;160  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$fc,d0;164  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$2,d0;168  00 00 00 02  ori.b	#$2, d0
      - ori.b	#$0,d0;16c  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$30,d0;170  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$fc,d0;174  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$1,d0;178  00 00 00 01  ori.b	#$1, d0
      - ori.b	#$80,d0;17c  00 00 00 80  ori.b	#$80, d0
      - ori.b	#$30,d0;180  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$fc,d0;184  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$3,d0;188  00 00 00 03  ori.b	#$3, d0
      - ori.b	#$49,d0;18c  00 00 00 49  ori.b	#$49, d0
      - ori.b	#$30,d0;190  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$fc,d0;194  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$2b,d0;198  00 00 00 2b  ori.b	#$2b, d0
      - ori.b	#$7,d0;19c  00 00 00 07  ori.b	#$7, d0
      - ori.b	#$30,d0;1a0  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$fc,d0;1a4  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$ff,d0;1a8  00 00 00 ff  ori.b	#$ff, d0
      - ori.b	#$fe,d0;1ac  00 00 00 fe  ori.b	#$fe, d0
      - ori.b	#$30,d0;1b0  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$fc,d0;1b4  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$1,d0;1b8  00 00 00 01  ori.b	#$1, d0
      - ori.b	#$80,d0;1bc  00 00 00 80  ori.b	#$80, d0
      - ori.b	#$30,d0;1c0  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$fc,d0;1c4  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$5,d0;1c8  00 00 00 05  ori.b	#$5, d0
      - ori.b	#$6c,d0;1cc  00 00 00 6c  ori.b	#$6c, d0
      - ori.b	#$30,d0;1d0  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$fc,d0;1d4  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$2c,d0;1d8  00 00 00 2c  ori.b	#$2c, d0
      - ori.b	#$7,d0;1dc  00 00 00 07  ori.b	#$7, d0
      - ori.b	#$30,d0;1e0  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$fc,d0;1e4  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$ff,d0;1e8  00 00 00 ff  ori.b	#$ff, d0
      - ori.b	#$fe,d0;1ec  00 00 00 fe  ori.b	#$fe, d0
      - ori.b	#$30,d0;1f0  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$fc,d0;1f4  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$1,d0;1f8  00 00 00 01  ori.b	#$1, d0
      - ori.b	#$80,d0;1fc  00 00 00 80  ori.b	#$80, d0
      - ori.b	#$30,d0;200  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$fc,d0;204  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$1,d0;208  00 00 00 01  ori.b	#$1, d0
      - ori.b	#$13,d0;20c  00 00 00 13  ori.b	#$13, d0
      - ori.b	#$30,d0;210  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$3c,d0;214  00 00 00 3c  ori.b	#$3c, d0
      - ori.b	#$0,d0;218  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$9,d0;21c  00 00 00 09  ori.b	#$9, d0
      - ori.b	#$32,d0;220  00 00 00 32  ori.b	#$32, d0
      - ori.b	#$3c,d0;224  00 00 00 3c  ori.b	#$3c, d0
      - ori.b	#$0,d0;228  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$50,d0;22c  00 00 00 50  ori.b	#$50, d0
      - ori.b	#$36,d0;230  00 00 00 36  ori.b	#$36, d0
      - ori.b	#$3c,d0;234  00 00 00 3c  ori.b	#$3c, d0
      - ori.b	#$80,d0;238  00 00 00 80  ori.b	#$80, d0
      - ori.b	#$7,d0;23c  00 00 00 07  ori.b	#$7, d0
      - ori.b	#$23,d0;240  00 00 00 23  ori.b	#$23, d0
      - ori.b	#$c8,d0;244  00 00 00 c8  ori.b	#$c8, d0
      - ori.b	#$0,d0;248  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$0,d0;24c  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$1,d0;250  00 00 00 01  ori.b	#$1, d0
      - ori.b	#$90,d0;254  00 00 00 90  ori.b	#$90, d0
      - ori.b	#$30,d0;258  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$c3,d0;25c  00 00 00 c3  ori.b	#$c3, d0
      - ori.b	#$30,d0;260  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$fc,d0;264  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$ff,d0;268  00 00 00 ff  ori.b	#$ff, d0
      - ori.b	#$fe,d0;26c  00 00 00 fe  ori.b	#$fe, d0
      - ori.b	#$30,d0;270  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$fc,d0;274  00 00 00 fc  ori.b	#$fc, d0
      - ori.b	#$1,d0;278  00 00 00 01  ori.b	#$1, d0
      - ori.b	#$80,d0;27c  00 00 00 80  ori.b	#$80, d0
      - ori.b	#$30,d0;280  00 00 00 30  ori.b	#$30, d0
      - ori.b	#$c1,d0;284  00 00 00 c1  ori.b	#$c1, d0
      - ori.b	#$d6,d0;288  00 00 00 d6  ori.b	#$d6, d0
      - ori.b	#$7c,d0;28c  00 00 00 7c  ori.b	#$7c, d0
      - ori.b	#$1,d0;290  00 00 00 01  ori.b	#$1, d0
      - ori.b	#$0,d0;294  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$d2,d0;298  00 00 00 d2  ori.b	#$d2, d0
      - ori.b	#$7c,d0;29c  00 00 00 7c  ori.b	#$7c, d0
      - ori.b	#$0,d0;2a0  00 00 00 00  ori.b	#$0, d0
      - ori.b	#$10,d0;2a4  00 00 00 10  ori.b	#$10, d0
      - ori.b	#$51,d0;2a8  00 00 00 51  ori.b	#$51, d0
      - ori.b	#$c8,d0;2ac  00 00 00 c8  ori.b	#$c8, d0
      - ori.b	#$ff,d0;2b0  00 00 00 ff  ori.b	#$ff, d0
      - ori.b	#$ea,d0;2b4  00 00 00 ea  ori.b	#$ea, d
      + movea.l	$4.w,a6; 0  2c 78 00 04  movea.l	$4.w, a6
      + moveq	#$0,d0; 4  70 00  moveq	#$0, d0
      + move.l	#$3e8,d0; 6  20 3c 00 00 03 e8  move.l	#$3e8, d0
      + move.l	#$10002,d1; c  22 3c 00 01 00 02  move.l	#$10002, d1
      + movea.l	$4.w,a6;12  2c 78 00 04  movea.l	$4.w, a6
      + jsr	-$c6(a6);16  4e ae ff 3a  jsr	-$c6(a6)
      + move.l	d0,$19c.l;1a  23 c0 00 00 01 9c  move.l	d0, $19c.l
      + lea.l	$17e(pc),a1;20  43 fa 01 5c  lea.l	$17e(pc), a1
      + jsr	-$198(a6);24  4e ae fe 68  jsr	-$198(a6)
      + movea.l	d0,a1;28  22 40  movea.l	d0, a1
      + move.l	$26(a1),d4;2a  28 29 00 26  move.l	$26(a1), d4
      + jsr	-$19e(a6);2e  4e ae fe 62  jsr	-$19e(a6)
      + move.b	#$80,d7;32  1e 3c 00 80  move.b	#$80, d7
      + move.w	#$ffff,d6;36  3c 3c ff ff  move.w	#$ffff, d6
      + move.w	$dff01c.l,d5;3a  3a 39 00 df f0 1c  move.w	$dff01c.l, d5
      + move.w	#$7fff,$dff09a.l;40  33 fc 7f ff 00 df f0 9a  move.w	#$7fff, $dff09a.l
      + movea.l	$19c(pc),a0;48  20 7a 01 52  movea.l	$19c(pc), a0
      + move.w	#$1fc,(a0)+;4c  30 fc 01 fc  move.w	#$1fc, (a0)+
      + move.w	#$0,(a0)+;50  30 fc 00 00  move.w	#$0, (a0)+
      + move.w	#$100,(a0)+;54  30 fc 01 00  move.w	#$100, (a0)+
      + move.w	#$200,(a0)+;58  30 fc 02 00  move.w	#$200, (a0)+
      + move.w	#$180,(a0)+;5c  30 fc 01 80  move.w	#$180, (a0)+
      + move.w	#$349,(a0)+;60  30 fc 03 49  move.w	#$349, (a0)+
      + move.w	#$2b07,(a0)+;64  30 fc 2b 07  move.w	#$2b07, (a0)+
      + move.w	#$fffe,(a0)+;68  30 fc ff fe  move.w	#$fffe, (a0)+
      + move.w	#$180,(a0)+;6c  30 fc 01 80  move.w	#$180, (a0)+
      + move.w	#$56c,(a0)+;70  30 fc 05 6c  move.w	#$56c, (a0)+
      + move.w	#$2c07,(a0)+;74  30 fc 2c 07  move.w	#$2c07, (a0)+
      + move.w	#$fffe,(a0)+;78  30 fc ff fe  move.w	#$fffe, (a0)+
      + move.w	#$180,(a0)+;7c  30 fc 01 80  move.w	#$180, (a0)+
      + move.w	#$113,(a0)+;80  30 fc 01 13  move.w	#$113, (a0)+
      + move.w	#$9,d0;84  30 3c 00 09  move.w	#$9, d0
      + move.w	#$50,d1;88  32 3c 00 50  move.w	#$50, d1
      + move.w	#$8007,d3;8c  36 3c 80 07  move.w	#$8007, d3
      + move.l	a0,$190.l;90  23 c8 00 00 01 90  move.l	a0, $190.l
      + move.w	d3,(a0)+;96  30 c3  move.w	d3, (a0)+
      + move.w	#$fffe,(a0)+;98  30 fc ff fe  move.w	#$fffe, (a0)+
      + move.w	#$180,(a0)+;9c  30 fc 01 80  move.w	#$180, (a0)+
      + move.w	d1,(a0)+;a0  30 c1  move.w	d1, (a0)+
      + add.w	#$100,d3;a2  d6 7c 01 00  add.w	#$100, d3
      + add.w	#$10,d1;a6  d2 7c 00 10  add.w	#$10, d1
      + dbra	d0,$96;aa  51 c8 ff ea  dbra	d0, $96
      + move.w	#$9,d0;ae  30 3c 00 09  move.w	#$9, d0
      + move.w	d3,(a0)+;b2  30 c3  move.w	d3, (a0)+
      + move.w	#$fffe,(a0)+;b4  30 fc ff fe  move.w	#$fffe, (a0)+
      + move.w	#$180,(a0)+;b8  30 fc 01 80  move.w	#$180, (a0)+
      + move.w	d1,(a0)+;bc  30 c1  move.w	d1, (a0)+
      + add.w	#$100,d3;be  d6 7c 01 00  add.w	#$100, d3
      + sub.w	#$10,d1;c2  92 7c 00 10  sub.w	#$10, d1
      + dbra	d0,$b2;c6  51 c8 ff ea  dbra	d0, $b2
      + move.l	a0,$198.l;ca  23 c8 00 00 01 98  move.l	a0, $198.l
      + move.w	d3,(a0)+;d0  30 c3  move.w	d3, (a0)+
      + move.w	#$fffe,(a0)+;d2  30 fc ff fe  move.w	#$fffe, (a0)+
      + move.w	#$180,(a0)+;d6  30 fc 01 80  move.w	#$180, (a0)+
      + move.w	#$113,(a0)+;da  30 fc 01 13  move.w	#$113, (a0)+
      + move.w	#$ffdf,(a0)+;de  30 fc ff df  move.w	#$ffdf, (a0)+
      + move.w	#$fffe,(a0)+;e2  30 fc ff fe  move.w	#$fffe, (a0)+
      + move.w	#$2c07,(a0)+;e6  30 fc 2c 07  move.w	#$2c07, (a0)+
      + move.w	#$fffe,(a0)+;ea  30 fc ff fe  move.w	#$fffe, (a0)+
      + move.w	#$180,(a0)+;ee  30 fc 01 80  move.w	#$180, (a0)+
      + move.w	#$56c,(a0)+;f2  30 fc 05 6c  move.w	#$56c, (a0)+
      + move.w	#$2d07,(a0)+;f6  30 fc 2d 07  move.w	#$2d07, (a0)+
      + move.w	#$fffe,(a0)+;fa  30 fc ff fe  move.w	#$fffe, (a0)+
      + move.w	#$180,(a0)+;fe  30 fc 01 80  move.w	#$180, (a0)+
      + move.w	#$349,(a0)+;102  30 fc 03 49  move.w	#$349, (a0)+
      + move.w	#$ffdf,(a0)+;106  30 fc ff df  move.w	#$ffdf, (a0)+
      + move.w	#$fffe,(a0)+;10a  30 fc ff fe  move.w	#$fffe, (a0)+
      + move.l	#$fffffffe,(a0);10e  20 bc ff ff ff fe  move.l	#$fffffffe, (a0)
      + move.l	$19c(pc),$dff080.l;114  23 fa 00 86 00 df f0 80  move.l	$19c(pc), $dff080.l
      + clr.w	$dff088.l;11c  42 79 00 df f0 88  clr.w	$dff088.l
      + moveq	#$50,d2;122  74 50  moveq	#$50, d2
      + neg.w	d6;124  44 46  neg.w	d6
      + move.w	#$c,d0;126  30 3c 00 0c  move.w	#$c, d0
      + bsr.w	$16e;12a  61 00 00 42  bsr.w	$16e
      + add.w	d6,d7;12e  de 46  add.w	d6, d7
      + dbra	d2,$136;130  51 ca 00 04  dbra	d2, $136
      + bra.b	$122;134  60 ec  bra.b	$122
      + move.w	#$13,d0;136  30 3c 00 13  move.w	#$13, d0
      + move.w	d7,d3;13a  36 07  move.w	d7, d3
      + movea.l	$190(pc),a3;13c  26 7a 00 52  movea.l	$190(pc), a3
      + move.b	d3,(a3);140  16 83  move.b	d3, (a3)
      + addq.w	#$1,d3;142  52 43  addq.w	#$1, d3
      + addq.w	#$6,a3;144  5c 4b  addq.w	#$6, a3
      + addq.w	#$2,a3;146  54 4b  addq.w	#$2, a3
      + dbra	d0,$140;148  51 c8 ff f6  dbra	d0, $140
      + movea.l	$198(pc),a3;14c  26 7a 00 4a  movea.l	$198(pc), a3
      + move.b	d3,(a3);150  16 83  move.b	d3, (a3)
      + btst.b	#$6,$bfe001.l;152  08 39 00 06 00 bf e0 01  btst.b	#$6, $bfe001.l
      + bne.b	$126;15a  66 ca  bne.b	$126
      + move.l	d4,$dff080.l;15c  23 c4 00 df f0 80  move.l	d4, $dff080.l
      + or.w	#$c000,d5;162  8a 7c c0 00  or.w	#$c000, d5
      + move.w	d5,$dff09a.l;166  33 c5 00 df f0 9a  move.w	d5, $dff09a.l
      + rts	;16c  4e 75  rts	
      + move.l	$dff004.l,d1;16e  22 39 00 df f0 04  move.l	$dff004.l, d1
      + lsr.l	#$1,d1;174  e2 89  lsr.l	#$1, d1
      + lsr.w	#$7,d1;176  ee 49  lsr.w	#$7, d1
      + cmp.w	d0,d1;178  b2 40  cmp.w	d0, d1
      + bne.b	$16e;17a  66 f2  bne.b	$16e
      + rts	;17c  4e 75  rts	
      + beq.b	$1f2;17e  67 72  beq.b	$1f2
      + bsr.b	$1f2;180  61 70  bsr.b	$1f2
      + bvc.b	$1ed;182  68 69  bvc.b	$1ed
      + bls.b	$1f9;184  63 73  bls.b	$1f9
      + movea.l	$6962(a4),a7;186  2e 6c 69 62  movea.l	$6962(a4), a7
      + moveq	#$61,d1;18a  72 61  moveq	#$61, d1
      + moveq	#$79,d1;18c  72 79  moveq	#$79, d1
      + ori.b	#$aa,d0;18e  00 00  ori.b	#$aa, d0
      +;
      
      [mocha] output truncated to 8192 characters, see "maxDiffSize" reporter-option

  	at Context.<anonymous> (d:\a\vscode-amiga-assembly\vscode-amiga-assembly\out\test\suite\extension.test.js:88:88)
  	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)

**Reply by grahambates** [reply]
I've opened a PR with a fix for the file disassembly issue.

**Reply by prb28** [reply]
Thanks !

**Comment by prb28** [comment]
@grahambates , I'm really sorry for the delay, but I finally took time to check the fs-uae on linux and Osx.
It took me some time to figure that the integration tests were failing upon some missing libraries.
Can you, please, add the same libraries for linux and osx as you did for the bartman amiga debug plugin ? (I can do a PR to your project if you wish).
Out of that the only issue I've seen is that on ubuntu the stop button does not kill the fs-uae process.

I've still some work to do to update the documentation and check the backward configuration compatibility and create a migration documentation.

**Comment by grahambates** [comment]
Sure, I'll add those. I'll do some more testing on my side too.

**Comment by WDrijver** [comment]
Hi Graham and Paul,
First of all my thanks to both of you for all the great work.
I work a lot with the Amiga Assembler in VSC and the debugging with FS-UAE is super.
I have almost finished an customised add-on for Contextual Doc for all Apollo SAGA registers and AMMX opcodes. 
I am very much looking forward to the unified GDBServer solution enabling the use of more recent versions of FS-UAE.
Willem Drijver
Apollo Team

**Comment by prb28** [comment]
Hi Willem, 
The last commit from Graham seems great.
I've some tests and documentation to prepare.
My hope is to publish it in beta before the end of this month.

**Comment by prb28** [comment]
@grahambates with the last fs-uae F5 (continue command) on my example code never hits breakpoint set on line 40 of gencop.s.

**Comment by grahambates** [comment]
Oh dear, I'll have a look.

**Reply by prb28** [reply]
Sorry about that, I have to prepare a debug environment with the new fs-uae code to try to do it by myself. But these two emulators doesn't build easily 😢

**Comment by grahambates** [comment]
It seems to only happen on Windows. I'll see if I can get to the bottom of it.

**Comment by grahambates** [comment]
I'm still not 100% sure what was going wrong, but switching back to my previous branch fixes this for now. I still kept the improved portable build script.

Some background:
In my FS-UAE fork for for Bartman's extension, I had to bring it up to date with the latest WinUAE. Even the latest dev build of FS-UAE is now pretty far behind, and the profiler relies on some internal data structures that have changed. This involved backporting the changes myself, and was a pretty major update to the codebase. Somewhere along the way it looks like this bug has been introduced. For Bartman's extension FS-UAE isn't used at all on Windows so this hadn't come up.

The good news is that for now, we're not using the profiler data in this extension so we can go back to the older branch based without the backported changes.

**Reply by prb28** [reply]
I've done a back-port from Winuae in my old version of fs-uae and it was not easy. Working on 3 different os is really painful.
I did, a quick test, it looks ok.
I'm struggling with some strange behavior of the windows runner of github actions, my integrations tests are failing.
